### ISSUE TITLE
Pieces for Developers snippet discovery crashes the client when looking at folders

### ISSUE THREAD

**Issue by twilwa** [pieces-app/documentation#95]
Environment: Win11 home, using Windows installation looking at Windows filesystem. There is another installation of Pieces on the machine in the wsl filesystem, which may be causing a conflict of some kind.
Expected Behavior: Snippet Discovery should generate snippets from files in a folder.
Actual Behavior: Snippet Discovery crashes the client without an error message when attempting to discover snippets from a folder.
It was able to successfully generate snippets from a folder on one occasion in the current session, but now seems to crash on any folder after attempting to restart Pieces OS/Desktop in order to troubleshoot.

![image](https://github.com/pieces-app/documentation/assets/111034762/488b588c-774d-42c8-9bfe-8a9053ee77a5)

**Comment by jordan-pieces** [comment]
hey @twilwa thanks for sharing this info and so sorry that you are having these issues with Snippet discovery.

Can you send over the version number of Pieces OS and Pieces for Developers when you were encountering this issue? And if you are still having this problem we can (hopefully) get this patched up for you asap! 

Send over those version numbers and let me know, we will test this here and see what we can get together. 

I have a few other questions if you can write back on any that you can answer: 

- What was the folder that you were attempting to perform snippet discovery on initially ?
- (optional) if the folder is small enough and does not contain sensitive information, would you be willing to share that directory for our testing?
- When you restarted the application, did you restart both Pieces OS and Pieces for Developers Desktop App ? (If you use task manager to close them both you can ensure that they have been quit entirely this way.)

Looking forward to hearing back and let us know if you have any of the above information and are willing to share!

**Comment by twilwa** [comment]
[autogpt.zip](https://github.com/pieces-app/documentation/files/12148681/autogpt.zip)
This is, if I recall correctly, the first folder I attempted to discover. I beleive this is the first one that failed -- the one that succeeded is an older version of babyagi that I've already updated since. After a restart this morning, I attempted to discover the same file again -- while it didn't hang and crash immediately, I switched focus windows while it was discovering, and found that the window was no longer open (assuming it crashed after all.). After reopening pieces for developers, the crash behavior I experienced yesterday is back.
OS_ 5.1.3 for developers 2.2.2
